frontend/src/admin/user-form.tsx

```diff
@@ -0,0 +1,142 @@
+import { User } from "zero-zummon";
+import { redirectToLogin } from "./use-users";
+
+export const UserForm: FunctionComponent = () => {
+    const {userId} = useParams<{ userId: string }>();
+    const [user, setUser] = useState<User | null>(null);
+    const [originalData, setOriginalData] = useState<User | null>(null);
+
+    const [loading, setLoading] = useState(false);
+    const [isEditing, setIsEditing] = useState(false);
+    const navigate = useNavigate();
+
+    const handleCancel = () => {
+        if (originalData) {
+            setUser(originalData); // Revert to original data
+        }
+        setIsEditing(false);
+    };
+
+    const handleEditToggle = () => {
+        setIsEditing(true);
+    };
+
+    const handleInputChange =(e: React.ChangeEvent<HTMLInputElement>) => {
+        const { name, value, type, checked } = e.target;
+        setUser((prev) => ({...prev,
+            [name]: type === "checkbox" ? checked : value,
+        } as User));
+    };
+
+    useEffect(() => {
+        if (userId) {
+            const fetchUser = async () => {
+                setLoading(true);
+                try {
+                    const response = await fetch(`${import.meta.env.VITE_ZTOR_URL}/users/${userId}`, {
+                        credentials: "include",
+                    });
+                    if (response.status === 401) {
+                        redirectToLogin();
+                        return;
+                    }
+                    if (response.ok) {
+                        const userData = await response.json();
+                        setUser(userData);
+                        setOriginalData(userData);
+                    } else {
+                        alert(`Error fetching user: ${response.statusText}`);
+                    }
+                } catch (error) {
+                    alert((error as Error).message);
+                } finally {
+                    setLoading(false);
+                }
+            };
+            fetchUser();
+        } else {
+            setIsEditing(true);
+        }
+    }, [userId]);
+
+    const handleSubmit = async (event: FormEvent) => {
+        event.preventDefault();
+        setLoading(true);
+        try {
+            const method = userId ? "PUT" : "POST";
+            const url = `${import.meta.env.VITE_ZTOR_URL}/users`
+            const response = await fetch(url, {
+                method,
+                headers: {
+                    "Content-Type": "application/json",
+                },
+                credentials: "include",
+                body: JSON.stringify(user),
+            });
+            if (response.status === 401) {
+                redirectToLogin();
+                return;
+            }
+
+            if (response.ok) {
+                navigate(`/users`);
+            } else {
+                const errorData = await response.json();
+                alert(`Error: ${errorData.message}`);
```

Comment:
In case of 403 I'm not seeing this alert because the response "Access denied" is plaintext instead of json